sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCOptions.scala

```diff
@@ -65,13 +65,38 @@ class JDBCOptions(
   // Required parameters
   // ------------------------------------------------------------
   require(parameters.isDefinedAt(JDBC_URL), s"Option '$JDBC_URL' is required.")
-  require(parameters.isDefinedAt(JDBC_TABLE_NAME), s"Option '$JDBC_TABLE_NAME' is required.")
+
   // a JDBC URL
   val url = parameters(JDBC_URL)
-  // name of table
-  val table = parameters(JDBC_TABLE_NAME)
+  val tableName = parameters.get(JDBC_TABLE_NAME)
```

Comment:
Personally I prefer:
```
  val tableExpression = if (parameters.isDefinedAt(JDBC_TABLE_NAME)) {
    require(!parameters.isDefinedAt(JDBC_QUERY_STRING), "...")
    parameters.get(JDBC_TABLE_NAME).get.trim
  } else {
    require(parameters.isDefinedAt(JDBC_QUERY_STRING), "...")
    s"(${parameters.get(JDBC_QUERY_STRING)}) ${curId.getAndIncrement()}"
  }
```

Reply:
@gengliangwang Thanks.. Actually i had tried a couple of different ways. Some how i found this a little hard to follow when i embed the error message. I like to check things upfront along with comments on top easy to follow. But if others find this easy to follow as well, then i will change.
```SQL
val tableExpression = if (parameters.isDefinedAt(JDBC_TABLE_NAME)) {
    require(!parameters.isDefinedAt(JDBC_QUERY_STRING),
      s"Both '$JDBC_TABLE_NAME' and '$JDBC_QUERY_STRING' can not be specified."
    )
    parameters.get(JDBC_TABLE_NAME).get.trim
  } else {
    require(parameters.isDefinedAt(JDBC_QUERY_STRING),
      s"Option '$JDBC_TABLE_NAME' or '${JDBC_QUERY_STRING}' is required."
    )
    s"(${parameters.get(JDBC_QUERY_STRING)}) ${curId.getAndIncrement()}"
  }
```